src/SimpleCalculator.Web/Controllers/HomeController.cs

```diff
@@ -1,48 +1,76 @@
 ﻿using System.Diagnostics;
+using AutoMapper;
+using SimpleCalculator.Web.Services.Interfaces;
 using Microsoft.AspNetCore.Mvc;
 using SimpleCalculator.Web.Models;
-using Calculations;
+using SimpleCalculator.DataAccess.Data;
 
 namespace SimpleCalculator.Web.Controllers;
 
 public class HomeController : Controller
 {
-    private readonly IAddition _addition;
-    private readonly ISubtraction _subtraction;
-    private readonly IMultiplication _multiplication;
-    private readonly IDivision _division;
+    private readonly ICalculationService<CalculationInputModel> _service;
+    private readonly CalculatorDbContext _context;
 
-    public HomeController(IAddition addition, ISubtraction subtraction, IMultiplication multiplication, IDivision division)
+    public HomeController(ICalculationService<CalculationInputModel> service, CalculatorDbContext context)
     {
-        _addition = addition;
-        _subtraction = subtraction;
-        _multiplication = multiplication;
-        _division = division;
+        _service = service;
+        _context = context;
     }
 
     public IActionResult Index()
     {
-        return View();
+        var e = _context.CalculationResultEntities.ToList();
+
+        var pModel = new CalculationPageModel()
+        {
+            Input = new CalculationInputModel(),
+            Result = new List<CalculationEntityModel>(),
+        };
+
+        
+        foreach (var entity in e)
```

Comment:
if you want to convert one model to another explicitly via `foreach`, then you can use `pModel.Result.Add` method and add an instance of `CalculationEntityModel` there.